engine/flink/executor/src/main/scala/pl/touk/nussknacker/engine/process/registrar/FlinkProcessRegistrar.scala

```diff
@@ -132,11 +143,18 @@ class FlinkProcessRegistrar(
       val exceptionHandlerPreparer = (runtimeContext: RuntimeContext) =>
         compilerDataForProcessPart(None)(runtimeContext.getUserCodeClassLoader).prepareExceptionHandler(runtimeContext)
       val jobData = compilerData.jobData
+
       FlinkCustomNodeContext(
         jobData,
         nodeComponentId.nodeId,
         compilerData.processTimeout,
-        convertToEngineRuntimeContext = FlinkEngineRuntimeContextImpl(jobData, _),
+        // TODO: this has to be changed
+        convertToEngineRuntimeContext = eng => {
```

Comment:
In proposed version in every place where `FlinkEngineRuntimeContextImpl` is used, `withProperMetricsProvider` method is called. This creates `FlinkEngineRuntimeContextImpl` with proper `MetricsProviderForScenario` (dummy provider for test, normal one otherwise).

This works everywhere apart from here in `FlinkProcessRegistrar`. Here, any use of `componentUseCase` results in `org.apache.flink.api.common.InvalidProgramException: The implementation of the RichMapFunction is not serializable` exception